dev/Breadcrumb/Breadcrumb.cpp

```diff
@@ -0,0 +1,561 @@
+    {
+        newLastItemImpl->SetPropertiesForLastNode();
+        m_lastBreadcrumbItem.set(newLastBreadcrumbItem);
+    }
+}
+
+void Breadcrumb::OnElementPreparedEvent(const winrt::ItemsRepeater&, const winrt::ItemsRepeaterElementPreparedEventArgs& args)
+{
+    if (const auto& item = args.Element().try_as<winrt::BreadcrumbItem>())
+    {
+        if (const auto& itemImpl = winrt::get_self<BreadcrumbItem>(item))
+        {
+            // The first element is always the ellipsis item
+            itemImpl->SetItemsRepeater(*this);
+
+            const uint32_t itemIndex = args.Index();
+            if (itemIndex == 0)
+            {
+                itemImpl->SetPropertiesForEllipsisNode();
+                itemImpl->SetFlyoutDataTemplate(DropdownItemTemplate());
+
+                m_ellipsisBreadcrumbItem.set(item);
+            }
+            else
+            {
+                const auto& itemsSourceAsList = this->ItemsSource().as<winrt::Collections::IVector<winrt::IInspectable>>();
+                const uint32_t itemCount = itemsSourceAsList.Size();
+
+                if (itemIndex == itemCount)
+                {
+                    UpdateLastElement(item);
+                }
+                else
+                {
+                    // Any other element just resets the visual properties
+                    itemImpl->ResetVisualProperties();
+                }
+            }
+        }
+    }
+}
+
+void Breadcrumb::OnElementIndexChangedEvent(const winrt::ItemsRepeater&, const winrt::ItemsRepeaterElementIndexChangedEventArgs& args)
+{
+    if (auto const element = args.Element())
+    {
+        element.SetValue(winrt::AutomationProperties::PositionInSetProperty(), box_value(args.NewIndex() + 1));
+
+        // When the selected item's index changes, update selection to match
+        if (auto const elementAsToggle = element.try_as<winrt::ToggleButton>())
+        {
+            if (SharedHelpers::IsTrue(elementAsToggle.IsChecked()))
+            {
+                FocusElement(args.NewIndex());
+            }
+        }
+    }
+}
+
+void Breadcrumb::OnElementClearingEvent(const winrt::ItemsRepeater&, const winrt::ItemsRepeaterElementClearingEventArgs& args)
+{
+    if (const auto& item = args.Element().try_as<winrt::BreadcrumbItem>())
+    {
+        const auto& itemImpl = winrt::get_self<BreadcrumbItem>(item);
+        itemImpl->ResetVisualProperties();
+        itemImpl->RevokeListeners();
+    }
+}
+
+winrt::IInspectable Breadcrumb::GenerateInternalItemsSource()
+{
+    // An internal ItemsSource is generated internally to add the
+    // ellipsis button
+    const auto& newItemsSource = winrt::make<Vector<IInspectable>>();
+
+    // The ellipsis button is created as a BreadcrumbItem, the element
+    // factory verifies that this element is already a BreadcrumbItem
+    // so no extra processing is needed
+    const auto& ellipsisItem = winrt::make<BreadcrumbItem>();
+    ellipsisItem.Content(winrt::box_value(L"..."));
```

Comment:
"..." should be a localized string resource.

Existing example:
tooltip.Content(box_value(ResourceAccessor::GetLocalizedStringResource(SR_TeachingTipAlternateCloseButtonTooltip)));

Reply:
> "..." should be a localized string resource.
> 
> Existing example:
> tooltip.Content(box_value(ResourceAccessor::GetLocalizedStringResource(SR_TeachingTipAlternateCloseButtonTooltip)));

Actually, shouldn't this be done as an Icon/Glyph?  There is a More Horizontal and More Vertical glyph used for overflow menu indicators.

![image](https://user-images.githubusercontent.com/7389110/103816654-3d909c00-505d-11eb-955a-db01cc2f34d0.png)

**MDL2** - E712

![image](https://user-images.githubusercontent.com/7389110/103816771-6add4a00-505d-11eb-8fc1-3059fa5ca9a7.png)

Reply:
Agree with @mdtauk here, having this be an icon makes a lot more sense then the literal string "...".

Reply:
UIA should be "**More**" rather than three full stops or an ellipses

Reply:
I think an E712 glyph is fine as long as it's appropriate for all localized languages.